### PR TITLE
NEP: Update NEP 47: Adopting the array API standard

### PR DESCRIPTION
**Pull request by asmeurer**
The NEP has been updated and expanded to more accurately correspond to the
implementation in #18585, as well as various wording and grammar fixes.

CC @rgommers @shoyer

### CODE REVIEW

File: doc/neps/nep-0047-array-api-standard.rst [hunk 1]
```diff
@@ -533,8 +627,11 @@ between many array libraries rather than only between NumPy and one other librar
 Alternatives
 ------------
 
-
-
+It was proposed to have the NumPy array API implementation as a separate
+library from NumPy. This was rejected because keeping it separate will make it
+less likely for people to review it, and including it in NumPy itself as an
+experimental submodule will make it easier for end users and library authors
+who already depend on NumPy to access the implementation.
```
Comment (asmeurer) [inline doc/neps/nep-0047-array-api-standard.rst:634]: @rgommers I noticed the "alternatives" section was empty, so I added a small paragraph. Did I miss any other motivations for including the implementation in NumPy, or other alternatives?

**Comment by charris** [comment]
Let's get this in. Thanks @asmeurer .

**Comment by shoyer** [comment]
Looks great to me, thank you @asmeurer !

**Comment by rgommers** [comment]
Looks like I'm late to the party here - just read it, looks really good. Thanks Aaron!

**Pull request merged**